src/main/java/Deadline.java

```diff
@@ -0,0 +1,59 @@
+import java.time.LocalDateTime;
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDateTime byDeadline;
+
+    /**
+     * Creates a deadline event.
+     * @param description task description
+     * @param byDeadline deadline for task
+     * @throws DukeException when there is a problem parsing the date time string provided by user
+     */
+    public Deadline(String description, String byDeadline) throws DukeException {
+        super(description);
+        this.taskType = TaskType.DEADLINE;
+        try {
+            this.byDeadline = LocalDateTime.parse(byDeadline, DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+        } catch (DateTimeParseException e) {
+            throw new DukeException("DateTime Parsing Failed: DateTime Format should follow \"dd/MM/yyyy HH:mm\" " 
+                    + "format." 
+                    + e.getMessage());
+        }
+    }
+
+    /**
+     * Converts DateTime to String in the format of dd/MM/yyyy HH:mm.
+     * @return DateTime as String in the format of dd/MM/yyyy HH:mm
+     */
+    public String getByDeadlineString() {

```

Comment:
Could it be a better idea to return a LocalDateTime and let the user of its return value decide how to format it? Let's say your task list is using it, it could decide on which pattern to display the date.